### ISSUE TITLE
[PWA] ngsw.json assetGroups are blank in Angular 12 (cause 504 error)

### ISSUE THREAD

**Issue by redplane** [angular/angular#42151]
# Bug Report

### Affected Package

 The issue is caused by package @angular/....

`"@angular/service-worker"`: "^12.0.0"

### Is this a regression?
- (Haven't tried with ealier version)

### Description
1. Follow instruction in [Angular official tutorial about PWA](https://angular.io/guide/service-worker-getting-started)
2. Host application using `http-server` (mentioned in the above tutorial)
3. Set chrome to **ONLINE** and load the page
4. Set chrome to **OFFLINE** and press **F5** to reload the loaded page

## Minimal Reproduction
- [Gitlab repository](https://gitlab.com/redplane/ngsw-playground)

## Exception or Error

![image](https://user-images.githubusercontent.com/7974095/118699506-cf928780-b83b-11eb-9f6e-ccaef1e45bdd.png)

## Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 12.0.0
Node: 14.17.0
Package Manager: npm 6.14.9
OS: win32 x64

Angular: 12.0.0
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker, upgrade

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0
@angular-devkit/build-angular   12.0.0
@angular-devkit/core            12.0.0
@angular-devkit/schematics      12.0.0
@schematics/angular             12.0.0
rxjs                            6.6.7
typescript                      4.2.4
</code></pre>

**Anything else relevant?**
What I notice is:
- In the generated `ngsw.json`, the `urls` in both `app` and `assets` section inside `assetGroups` are all blank (see the below image)

![image](https://user-images.githubusercontent.com/7974095/118699947-4cbdfc80-b83c-11eb-8806-959407ae3ebd.png)

- When I filled values into `urls` as the below image, the PWA worked perfectly. App can work in Offline mode

![image](https://user-images.githubusercontent.com/7974095/118700121-8131b880-b83c-11eb-80d5-2ae7bce39567.png)

**Comment by hamidkhandatamagnetics** [comment]
I am also having exact this same issue with service worker.

**Comment by Kashif-datamagnetics** [comment]
Experiencing same issue, just tried with a band new application and issue is producing .This is high priory issue and must be fixed quickly. Many thanks

**Comment by ghazanfardmt** [comment]
Its need an urgent action to fix. Facing same issue with NG-12.

**Comment by AfzalDMT** [comment]
Facing the same issue and looking for solution as soon as possible. Thanks

**Comment by nazim-hussain** [comment]
I am facing this issue with service worker, can you please look into it urgently, Thank you

**Comment by waqas780** [comment]
I am facing this issue with service worker, can you please look into it urgently, Thank you

**Comment by kashifsohail** [comment]
i am also facing same issue, need solution urgently

**Comment by SohailLatif** [comment]
I  am  same issue, just tried with a band new application and issue is producing .an you please look into it urgently, 
Thank you

**Comment by usmanghani-DMT** [comment]
Its need an urgent action to fix. Facing same issue

**Comment by raosafdar** [comment]
Facing Same Issue
desperately need Solution 
Thank You

**Comment by aqeel-dmt** [comment]
yes exactly facing same issue, need urgently solution on it. thanks.

**Comment by mtariqdmt** [comment]
Facing same issue and looking for it urgent solution.
